source/extensions/common/wasm/wasm.cc

```diff
@@ -1407,24 +1407,20 @@ void Wasm::registerCallbacks() {
       "env", #_fn, &Exports::_fn,                                                                  \
       &ConvertFunctionWordToUint32<decltype(Exports::_fn),                                         \
                                    Exports::_fn>::convertFunctionWordToUint32)
-  if (is_emscripten_) {
-    _REGISTER(pthread_equal);
-  }
+  _REGISTER(pthread_equal);
```

Comment:
Note: I remove the `is_emscripten_` guard here and for WASI because clones don't have access to Wasm code, and we don't copy this value to them... and while we could set `is_emscripten_` in clones, seeing as WASI is becoming a standard and other toolchains use it as well, I think it's fine for us to always expose those functions, so that modules can link against them, if needed.

Reply:
Agreed